deploy/src/utils/verifier.js

```diff
@@ -0,0 +1,102 @@
+const axios = require('axios')
+const querystring = require('querystring')
+const flattener = require('truffle-flattener')
+const { EXPLORER_TYPES, REQUEST_STATUS } = require('../constants')
+const promiseRetry = require('promise-retry')
+
+const flat = async contractPath => flattener([contractPath], process.cwd())
```

Comment:
do we need to flatten contracts one more time? We already have them flattened in `flats` directory. The idea is to use files from `flats` in order to make sure that deployed contracts can be built from them. My plan is to provide content of `flats` directory as a release artefacts.

Reply:
I thought about using the `flats` directory but here are some reasons why I chose to not use it:
- It will require that the `flatten` script is run before the deployment script, and it should be up-to-date in case of adding new contracts. For this, we can create a `deploy` npm script instead of running `node deploy.js` and call first the `flatten` script to make sure they are created. 
- In the implementation, we need to make a relation between the deployed contract and the flatten location. The truffle artifact provides the location of the source code and the script uses it to call truffle-flattener for that contract. In this case, I think we can calculate the location of the flatten contract from the source location, but it will be prone to break if any directory is changed

Reply:
Thanks for analysis!

> It will require that the flatten script is run before the deployment script

I have no objections for this.

Reply:
Updated the logic to use `flats` directory https://github.com/poanetwork/tokenbridge-contracts/pull/334/commits/04bb5785a6451fd09daa05d82e02de388dd2bc92

Reply:
thanks